proto/store/idp.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto3";
+
+package bytebase.store;
+option go_package = "generated-go/store";
+
+// FieldMapping uses for mapping user info from identity provider to Bytebase.
+message FieldMapping {
+  string identifier = 1;
+
+  string name = 2;
+
+  string email = 3;
+}
+
+// OAuth2IdentityProviderConfig is the structure for OAuth2 identity provider config.
+message OAuth2IdentityProviderConfig {
+  string authURL = 1;
+
+  string tokenURL = 2;
+
+  string userInfoURL = 3;
+
+  string clientID = 4;
+
+  string clientSecret = 5;
+
+  string scopes = 6;
```

Comment:
Shall this be repeated string?

Reply:
As we only put it in the request parameters, so I think the string type is enough.
e.g. `${user_info_url}?scopes=email%20profile`

Reply:
Better to be repeated string then if we can separate strings. In the worst case, we can use a singleton string.

Reply:
Updated